tools/donate_cpu_lib.py

```diff
@@ -210,46 +209,41 @@ def compile_cppcheck(cppcheck_path):
 
 def get_cppcheck_versions():
     print('Connecting to server to get Cppcheck versions..')
-    try:
-        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
-            sock.connect(__server_address)
-            sock.send(b'GetCppcheckVersions\n')
-            versions = sock.recv(256)
-    except socket.error as err:
-        print('Failed to get cppcheck versions: ' + str(err))
-        return None
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+        sock.connect(__server_address)
+        sock.send(b'GetCppcheckVersions\n')
+        versions = sock.recv(256)
+    # TODO: sock.recv() sometimes hangs and returns b'' afterwards
```

Comment:
This happens quite regularly with a local instance. The client hangs and the server does nothing until it returns empty data. Restarting the client has no impact so it appears to be a server-side issue - maybe even an issue in our code. Will take a closer look after I am done with my pending local changes.